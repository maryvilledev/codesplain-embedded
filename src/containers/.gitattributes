# Denote yarn.lock as binary file
yarn.lock binary
