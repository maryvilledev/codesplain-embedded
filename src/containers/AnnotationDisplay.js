import React, { Component, PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import { connect } from 'react-redux';

import markdownRendererOptions from '../util/markdown-renderer-options';

const styles = {
  container: {
    flex: '1 1 auto',
  },
};

class AnnotationDisplay extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      annotations,
      selectedLine,
    } = this.props;
    if (selectedLine === -1) {
      return (
        <div style={styles.container}>
          <h1>Annotation</h1>
        </div>
      );
    }

    const annotation = annotations[selectedLine + 1];
    return (
      <div style={styles.container}>
        <h1>Annotation</h1>
        <MarkdownRenderer
          markdown={annotation.annotation}
          options={markdownRendererOptions}
        />
      </div>
    );
  }
}

AnnotationDisplay.propTypes = {
  selectedLine: PropTypes.number,
}

AnnotationDisplay.defaultProps = {
  selectedLine: -1,
}

const mapStateToProps = (state) => ({
  annotations: state.annotations,
  selectedLine: state.selectedLine,
});

export default connect(mapStateToProps)(AnnotationDisplay);
