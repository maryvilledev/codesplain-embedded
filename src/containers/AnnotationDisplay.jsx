import React, { Component, PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import { connect } from 'react-redux';

import { setSelectedLine } from '../actions/app';
import {
  getAnnotatedLines,
  getNextAnnotation,
  getPreviousAnnotation,
  hasNextAnnotation,
  hasPreviousAnnotation,
} from '../util/annotations';
import markdownRendererOptions from '../util/markdown-renderer-options';

const styles = {
  container: {
    flex: '1 1 auto',
    flexBasis: '50%', // "default width"
    paddingLeft: '15px',
  },
  markdownContainer: {
    background: '#fff',
    padding: '10px',
    borderRadius: '5px',
  },
  buttonContainer: {
    display: 'inline-flex',
    justifyContent: 'flex-start',
  },
};

class AnnotationDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasPreceedingAnnotation: false,
      hasProceedingAnnotation: false,
    };
    this.getNextAnnotation = this.getNextAnnotation.bind(this);
    this.getPreviousAnnotation = this.getPreviousAnnotation.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const {
      annotations,
      selectedLine,
    } = nextProps;
    const nextAnnotatedLines = getAnnotatedLines(annotations);
    this.setState({
      hasProceedingAnnotation: hasNextAnnotation(nextAnnotatedLines, selectedLine),
      hasPreceedingAnnotation: hasPreviousAnnotation(nextAnnotatedLines, selectedLine),
    });
  }

  getPreviousAnnotation() {
    const {
      annotations,
      dispatch,
      selectedLine,
    } = this.props;
    // Get the annotated lines
    const previous = getPreviousAnnotation(annotations, selectedLine);
    if (!previous) {
      return;
    }
    dispatch(setSelectedLine(previous.lineNumber));
  }

  getNextAnnotation() {
    const {
      annotations,
      dispatch,
      selectedLine,
    } = this.props;
    // Get the annotated lines
    const next = getNextAnnotation(annotations, selectedLine);
    if (!next) {
      return;
    }
    dispatch(setSelectedLine(next.lineNumber));
  }

  render() {
    const {
      hasPreceedingAnnotation,
      hasProceedingAnnotation,
    } = this.state;
    const {
      annotations,
      selectedLine,
    } = this.props;
    if (selectedLine === -1) {
      return (
        <div style={styles.container}>
          <h2>Annotation</h2>
        </div>
      );
    }
    const annotation = annotations[selectedLine];
    return (
      <div style={styles.container}>
        <h2>Annotation</h2>
        <div style={styles.buttonContainer}>
          <button
            disabled={hasPreceedingAnnotation}
            onClick={this.getPreviousAnnotation}
            type="button"
          >
            Previous
          </button>
          <button
            disabled={hasProceedingAnnotation}
            onClick={this.getNextAnnotation}
            type="button"
          >
            Next
          </button>
        </div>
        <div style={styles.markdownContainer}>
          <MarkdownRenderer
            markdown={annotation.annotation}
            options={markdownRendererOptions}
          />
        </div>
      </div>
    );
  }
}

AnnotationDisplay.propTypes = {
  selectedLine: PropTypes.number,
}

AnnotationDisplay.defaultProps = {
  selectedLine: -1,
}

const mapStateToProps = (state) => ({
  annotations: state.annotations,
  selectedLine: state.selectedLine,
});

export default connect(mapStateToProps)(AnnotationDisplay);
