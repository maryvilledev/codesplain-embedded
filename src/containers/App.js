import React, {Component, PropTypes} from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

const mockReducer = (state={}, action) => state

class App extends Component {
  constructor(props) {
    super(props);
    this.store = createStore(mockReducer);
  }
  render() {
    return (
      <Provider store={this.store}>
        <div>Ello govna?</div>
      </Provider>
    );
  }
}

App.propTypes = {
  snippetKey: PropTypes.string.required,
}

export default App
