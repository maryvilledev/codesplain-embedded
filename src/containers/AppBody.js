import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

class AppBody extends Component {
  componentDidMount() {
    const { dispatch, snippetKey } = this.props;
    axios.get("../public/snippet.json")
      .then((res) => {
        console.log(res.data.snippetTitle)
      });
  }
  render() {
    return null;
  }
}

AppBody.propTypes = {
  snippetKey: PropTypes.string.required,
}

export default connect()(AppBody);
