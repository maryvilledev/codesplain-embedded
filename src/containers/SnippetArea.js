import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

class SnippetArea extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { snippet, title, language } = this.props;

    return (
      <div>
        <h1>{title}</h1>
        <h3>A {language} snippet</h3>
        <textarea
          value={snippet}
          readOnly={true}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  snippet:  state.snippet,
  title:    state.snippetTitle,
  language: state.snippetLanguage,
});

export default connect(mapStateToProps)(SnippetArea);
