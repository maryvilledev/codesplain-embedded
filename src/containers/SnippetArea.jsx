import React, { Component } from 'react';
import { connect } from 'react-redux';

import { setSelectedLine } from '../actions/app';

import Editor from '../components/Editor';

const styles = {
  container: {
    flex: '1 1 auto',
    flexBasis: '50%', // "default width"
    marginTop: '30px',
  },
};

export class SnippetArea extends Component {
  constructor(props) {
    super(props);
    this.handleGutterClick = this.handleGutterClick.bind(this);
  }

  handleGutterClick(line) {
    const { dispatch } = this.props;
    dispatch(setSelectedLine(line));
  }

  render() {
    const {
      snippet,
      language,
      annotations,
      AST,
      filters,
      openLine
    } = this.props;
    const markedLines = Object.keys(annotations).map(key => Number(key));

    return (
      <div style={styles.container}>
        <Editor
          snippet={snippet}
          markedLines={markedLines}
          onGutterClick={this.handleGutterClick}
          language={language}
          AST={AST}
          filters={filters}
          openLine={openLine}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  snippet:  state.snippet,
  language: state.snippetLanguage,
  annotations: state.annotations,
  AST: state.AST,
  filters: state.filters,
  openLine: state.selectedLine
});

export default connect(mapStateToProps)(SnippetArea);
