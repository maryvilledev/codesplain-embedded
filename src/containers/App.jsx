import React, {Component, PropTypes} from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import AppBody from './AppBody';
import appReducer from '../reducers/app';

class App extends Component {
  constructor(props) {
    super(props);
    this.store = createStore(appReducer);
  }
  render() {
    return (
      <Provider store={this.store}>
        <AppBody {...this.props} />
      </Provider>
    );
  }
}

export default App;
