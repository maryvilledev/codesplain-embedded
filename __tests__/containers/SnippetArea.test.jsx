import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import { SnippetArea } from '../../src/containers/SnippetArea';

describe('<SnippetArea />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <SnippetArea
        AST={{}}
        filters={{}}
        snippet="Wubba lubba Dub Dub"
        language="bad"
        openLine={0}
        onGutterClick={jest.fn()}
        annotations={[{"0": "They're just Robots!"}]}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
