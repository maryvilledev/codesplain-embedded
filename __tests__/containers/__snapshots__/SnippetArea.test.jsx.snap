// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnippetArea /> matches snapshot 1`] = `
<div
  style={
    Object {
      "flex": "1 1 auto",
      "flexBasis": "50%",
      "marginTop": "30px",
    }
  }
>
  <Editor
    AST={Object {}}
    filters={Object {}}
    language="bad"
    markedLines={
      Array [
        0,
      ]
    }
    onGutterClick={[Function]}
    openLine={0}
    snippet="Wubba lubba Dub Dub"
  />
</div>
`;
